### PR TITLE
Adding a way to access the number of epochs to instantiate lr schedulers

### PR DESCRIPTION
**Pull request by zaccharieramzi**
- LR schedulers depend on n epochs
- Shuffling in tf dataset (tackles #7 )
- slight perf improvement in pl module

### CODE REVIEW

File: utils/tf_solver.py [hunk 1]
```diff
@@ -51,15 +51,15 @@ def set_objective(self, model_init_fn, dataset, normalization):
                 tf.keras.optimizers.schedules.ExponentialDecay(
                     value,
                     decay_rate=0.1,
-                    decay_steps=30,
+                    decay_steps=n_epochs//2,
```
Comment (tomMoral) [inline utils/tf_solver.py:54]: didn't we say `[n_epochs//2, 3*n_epochs//4]]` to get 2 steps?

File: utils/tf_solver.py [hunk 2]
```diff
@@ -39,7 +39,7 @@ def skip(self, model_init_fn, dataset, normalization):
             return True, 'Cannot use both decoupled and coupled weight decay'
         return False, None
 
-    def set_objective(self, model_init_fn, dataset, normalization):
+    def get_lr_wd_cback(self, n_epochs=200):
```
Comment (tomMoral) [inline utils/tf_solver.py:42]: I would call this `max_epochs`, as we are not sur to reach it.

File: utils/tf_solver.py [hunk 3]
```diff
@@ -94,7 +97,10 @@ def set_objective(self, model_init_fn, dataset, normalization):
                 lambda x, y: (data_aug_layer(x[None], training=True)[0], y),
                 num_parallel_calls=tf.data.experimental.AUTOTUNE,
             )
-        self.dataset = self.dataset.batch(
+        self.dataset = self.dataset.shuffle(
+            buffer_size=1000,  # For now a hardcoded value
```
Comment (tomMoral) [inline utils/tf_solver.py:101]: can't this be auto tuned?
if not, seems fine like this 😅

**Review by tomMoral — APPROVED** [review]
LGTM.

a few nitpicks

File: utils/tf_solver.py (hunk 3 above)
Comment (zaccharieramzi) [reply utils/tf_solver.py:101]: Not possible unfortunately haha !

File: utils/tf_solver.py (hunk 2 above)
Comment (zaccharieramzi) [reply utils/tf_solver.py:42]: sure

File: utils/tf_solver.py (hunk 1 above)
Comment (zaccharieramzi) [reply utils/tf_solver.py:54]: so the thinking was that in order to do a minimal PR, since it's not possible to have this specific schedule I would forget it. But I will actually try to change it.

Comment (zaccharieramzi) [reply utils/tf_solver.py]: not that much of a change in the end haha!

File: objective.py [hunk 4]
```diff
@@ -160,7 +160,8 @@ def set_data(
         # https://pytorch-lightning.readthedocs.io/en/stable/common/trainer.html#reproducibility
         # XXX: modify this with the correct amount of CPUs/GPUs
         self.trainer = Trainer(
-            accelerator="auto", strategy="noteardown", max_epochs=-1
+            accelerator="auto", strategy="noteardown", max_epochs=-1,
+            enable_checkpointing=False,
```
Comment (tomMoral) [inline objective.py:164]:
```suggestion
            enable_checkpointing=False,
            enable_model_summary=False
```

**Review by tomMoral — APPROVED** [review]
LGTM
